### ISSUE TITLE
construction of a measurement function 

### ISSUE THREAD

**Issue by DikshantDulal** [mit-han-lab/torchquantum#56]
Can we construct a measurement function that takes q_device as input and gives the measurement outcomes of classical values as output? It really took me **2 days** to figure out how to obtain the distribution of classical outcomes upon measurement when I have q_device as an attribute of my class but not q_state. 

it would be something like the following: This might help the future users!

```
def measure_device(q_device, n_shots=1024):
    """
    Measures the q_device and returns the classical bitstream distribution
    """
    q_state = tq.QuantumState(n_wires=q_device.n_wires)
    q_state.set_states(q_device.get_states_1d())
    return q_state.measure(q_state, n_shots=n_shots)
```

Given properties of the task: The class that I was defining was MAXCUT for QAOA whose attributes look like this.
```
class MAXCUT(tq.QuantumModule):
    def __init__(self, n_wires, input_graph):
        super().__init__()
        self.n_wires = n_wires
        self.input_graph = input_graph  # list of edges
        self.q_device = tq.QuantumDevice(n_wires=n_wires)
        self.rx0 = tq.RX(has_params=True, trainable=True)
        self.rz0 = tq.RZ(has_params=True, trainable=True)
        self.measure = tq.MeasureAll(tq.PauliZ)
```

**Comment by Hanrui-Wang** [comment]
Hi Dikshant,

Thank you for raising this great point! I will add this to the documentation.

For the MAXCUT, seems the forward function is not shown here?

**Comment by Hanrui-Wang** [comment]
Actually, for obtaining classical bit string from tq.QuantumDevice, a simpler way is to directly use tq.measure:

`tq.measure(q_device, n_shots=1024)`

Please see README basic usage 2 for a complete example

**Comment by DikshantDulal** [comment]
Hi Han Rui, thank you! that actually resolves my problem! I wonder where this function was! Because in measurement.py which I was referring to,  the only measure function that I found was below which takes **q_state** as the input but not the **q_device**.

 `def measure(q_state, n_shots=1024, draw_id=None):
    """Measure the target state and obtain classical bitstream distribution

    Args:
        q_state: input tq.QuantumState
        n_shots: number of simulated shots
        draw_id: which state to draw

    Returns:
        distribution of bitstrings

    """
    bitstring_candidates = gen_bitstrings(q_state.n_wires)

    state_mag = q_state.get_states_1d().abs().detach().numpy()
    distri_all = []

    for state_mag_one in state_mag:
        state_prob_one = np.abs(state_mag_one) ** 2
        measured = random.choices(
            population=bitstring_candidates,
            weights=state_prob_one,
            k=n_shots,
        )
        counter = Counter(measured)
        counter.update({key: 0 for key in bitstring_candidates})
        distri = dict(counter)
        distri = OrderedDict(sorted(distri.items()))
        distri_all.append(distri)

    if draw_id is not None:
        plt.bar(distri_all[draw_id].keys(), distri_all[draw_id].values())
        plt.xticks(rotation='vertical')
        plt.xlabel('bitstring [qubit0, qubit1, ..., qubitN]')
        plt.title('distribution of measured bitstrings')
        plt.show()
    return distri_all`
